lib/matplotlib/axes/_axes.py

```diff
@@ -8572,6 +8599,24 @@ def violin(self, vpstats, positions=None, vert=True, widths=0.5,
         datashape_message = ("List of violinplot statistics and `{0}` "
                              "values must have the same length")
 
+        if vert is not None:
+            _api.warn_deprecated(
+                "3.10",
+                name="vert: bool",
+                alternative="orientation: {'vertical', 'horizontal'}"
+            )
+
+        # vert and orientation parameters are linked until vert's
+        # deprecation period expires. If both are selected,
+        # vert takes precedence.
+        if vert or vert is None and orientation is None:
+            orientation = 'vertical'
+        elif vert is False:
+            orientation = 'horizontal'
+
+        if orientation is not None:
+            _api.check_in_list(['horizontal', 'vertical'], orientation=orientation)
```

Comment:
I'm not following the logic here completely. After whatever mangling vert and orientation, shouldn't we have a normalized `orientation` that must be 'horizontal' or 'vertical'? In other words

- Why do we allow orientation to be None in 8617 and would that even work for the following code?

Can't we simply
```
if vert is not None:
    _api.warn_deprecated(...)
    orientation = 'vertical' if vert else 'horizontal'
_api.check_in_list(['horizontal', 'vertical'], orientation=orientation)
```
and default `orientation='vertical'` instead of defaulting it to None?

Reply:
Yes it should default to vertical instead of None (it even says so in the docstring) so there was no reason for it other than making it more convoluted. That code change is great, I'll add it to the next commit